scripts/graph_accelerometer.py

```diff
@@ -42,6 +39,126 @@ def plot_accel(data, logname):
     fig.tight_layout()
     return fig
 
+
+######################################################################
+# Frequency graphing
+######################################################################
+
+# Calculate estimated "power spectral density"
+def calc_freq_response(data, max_freq):
+    helper = ShaperCalibrate(printer=None)
+    return helper.process_accelerometer_data(data)
+
+def calc_specgram(data, axis):
+    N = data.shape[0]
+    Fs = N / (data[-1,0] - data[0,0])
+    # Round up to a power of 2 for faster FFT
+    M = 1 << int(.5 * Fs - 1).bit_length()
+    window = np.kaiser(M, 6.)
+    def _specgram(x):
+        return matplotlib.mlab.specgram(
+                x, Fs=Fs, NFFT=M, noverlap=M//2, window=window,
+                mode='psd', detrend='mean', scale_by_freq=False)
+
+    d = {'x': data[:,1], 'y': data[:,2], 'z': data[:,3]}
+    if axis != 'all':
+        pdata, bins, t = _specgram(d[axis])
+    else:
+        pdata, bins, t = _specgram(d['x'])
+        for ax in 'yz':
+            pdata += _specgram(d[ax])[0]
+    return pdata, bins, t
+
+def plot_frequency(datas, lognames, max_freq):
+    calibration_data = calc_freq_response(datas[0], max_freq)
+    for data in datas[1:]:
+        calibration_data.join(calc_freq_response(data, max_freq))
+    freqs = calibration_data.freq_bins
+    psd = calibration_data.psd_sum[freqs <= max_freq]
+    px = calibration_data.psd_x[freqs <= max_freq]
+    py = calibration_data.psd_y[freqs <= max_freq]
+    pz = calibration_data.psd_z[freqs <= max_freq]
+    freqs = freqs[freqs <= max_freq]
+
+    fig, ax = matplotlib.pyplot.subplots()
+    ax.set_title("\n".join(wrap(
+        "Frequency response (%s)" % (', '.join(lognames)), MAX_TITLE_LENGTH)))
+    ax.set_xlabel('Frequency (Hz)')
+    ax.set_ylabel('Power spectral density')
+
+    ax.plot(freqs, psd, label='X+Y+Z', alpha=0.6)
+    ax.plot(freqs, px, label='X', alpha=0.6)
+    ax.plot(freqs, py, label='Y', alpha=0.6)
+    ax.plot(freqs, pz, label='Z', alpha=0.6)
+
+    ax.xaxis.set_minor_locator(matplotlib.ticker.AutoMinorLocator())
+    ax.yaxis.set_minor_locator(matplotlib.ticker.AutoMinorLocator())
+    ax.xaxis.set_minor_locator(matplotlib.ticker.AutoMinorLocator())
+    ax.yaxis.set_minor_locator(matplotlib.ticker.AutoMinorLocator())
```

Comment:
Is the double setting (here and on line 121-124) actually correct?

Reply:
Yep, it was an accumulated copy-paste error, apparently.